File: frontend/types/banner.ts
export interface BannerRedirectConfig {
  enabled: boolean;
  type: 'none' | 'page' | 'product' | 'custom';
  value?: string | number;
  resolved_url?: string;
}

export interface BannerItem {
  imageUrl: string;
  altText?: string;
  redirectUrl?: string; // Keep for backward compatibility
  redirect?: BannerRedirectConfig;
  order: number;
}

export interface BannerResponsiveConfig {
  mobileContainerStyle?: string;
  desktopContainerStyle?: string;
  mobileGridColumns?: number;
}

export interface BannerFloatingPosition {
  anchor: 'top-left' | 'top-right' | 'center' | 'bottom-left' | 'bottom-right' | 'custom';
  offsetX?: number;
  offsetY?: number;
}

export interface BannerFloatingConfig {
  // Tương thích ngược với dữ liệu cũ
  anchor?: 'top-left' | 'top-right' | 'center' | 'bottom-left' | 'bottom-right' | 'custom';
  offsetX?: number;
  offsetY?: number;
  
  // Cấu hình định vị riêng cho từng thiết bị
  desktopPosition?: BannerFloatingPosition;
  tabletPosition?: BannerFloatingPosition;
  mobilePosition?: BannerFloatingPosition;

  closeable?: boolean;
  closeSession?: boolean;
  zIndex?: number;
  shadow?: string;
}

export interface BannerLayoutConfig {
  containerStyle?: string;
  animation?: string;
  items: BannerItem[];
  popupDelay?: number;
  showCloseButton?: boolean;
  gridColumns?: number;
  gridGap?: number;
  responsive?: BannerResponsiveConfig;
  floatingConfig?: BannerFloatingConfig;
}

export interface Banner {
  id: number;
  name: string;
  position: string;
  type: 'slideshow' | 'popup' | 'sidebar' | 'grid' | 'floating' | string;
  page?: string;
  status: 'Draft' | 'Published';
  version: string;
  layoutConfig: BannerLayoutConfig;
  draftConfig?: BannerLayoutConfig;
  hasUnpublishedChanges?: boolean;
  publishedAt?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateOrUpdateBannerRequest {
  name: string;
  position: string;
  type: string;
  layoutConfig: BannerLayoutConfig;
}

export interface BannerVersion {
  id: number;
  bannerId: number;
  version: string;
  layoutConfig: BannerLayoutConfig;
  createdAt: string;
  createdBy?: string;
}

File: backend/Models/Banner.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PolyBabyAPI.Models
{
    public class Banner
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        public string Name { get; set; } = string.Empty;

        [Required]
        [StringLength(50)]
        public string Position { get; set; } = string.Empty; // home, left, popup, banner_top

        [Required]
        [StringLength(50)]
        public string Type { get; set; } = string.Empty; // slideshow, popup, sidebar, grid

        [StringLength(20)]
        public string Status { get; set; } = "Draft"; // Draft, Published

        [Required]
        [StringLength(50)]
        public string Page { get; set; } = "global"; // global, home, products, product_detail, cart, checkout, profile

        [StringLength(50)]
        public string Version { get; set; } = "1.0.0";

        public BannerLayoutConfig LayoutConfig { get; set; } = new BannerLayoutConfig();

        public DateTime? PublishedAt { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime UpdatedAt { get; set; } = DateTime.Now;
        
        public string? CreatedBy { get; set; }
    }

    public class BannerLayoutConfig
    {
        // Common config
        public string? ContainerStyle { get; set; }
        public string? Animation { get; set; }
        
        // For components that have multiple items (slideshow, grid) or single item (popup, sidebar)
        public List<BannerItem> Items { get; set; } = new List<BannerItem>();
        
        // Popup specific
        public int? PopupDelay { get; set; } // in ms
        public bool? ShowCloseButton { get; set; }
        
        // Grid specific
        public int? GridColumns { get; set; }
        public int? GridGap { get; set; }
        
        // Responsive config
        public BannerResponsiveConfig? Responsive { get; set; } = new BannerResponsiveConfig();
    }

    public class BannerItem
    {
        public string ImageUrl { get; set; } = string.Empty;
        public string? AltText { get; set; }
        public string? RedirectUrl { get; set; }
        public int Order { get; set; }
    }

    public class BannerResponsiveConfig
    {
        public string? MobileContainerStyle { get; set; }
        public string? DesktopContainerStyle { get; set; }
        public int? MobileGridColumns { get; set; }
    }
}
